Content.Client/Disposal/Systems/DisposalUnitSystem.cs

```diff
@@ -128,10 +129,6 @@ private void UpdateState(EntityUid uid, SharedDisposalUnitComponent unit, Sprite
                             {
                                 // Play the flush animation
                                 new AnimationTrackSpriteFlick.KeyFrame(flushState, 0),
-                                // Return to base state (though, depending on how the unit is
-                                // configured we might get an appearance change event telling
-                                // us to go to charging state)
-                                new AnimationTrackSpriteFlick.KeyFrame(chargingState, (float) unit.FlushDelay.TotalSeconds)
```

Comment:
AnimationTrackSpriteFlick line 65 will overwrite the LayerKey state with each KeyFrame, so the state will be overwritten if we don't remove this second KeyFrame. I see there is a TODO to maybe allow multiple LayerKeys, which might avoid this issue.

I've tested with fake lag CVAR turned up and it has always played correctly for me.